core/include/traccc/edm/cluster.hpp

```diff
@@ -9,31 +9,46 @@
 
 // Project include(s).
 #include "traccc/edm/cell.hpp"
+#include "traccc/geometry/pixel_data.hpp"
 
 // System include(s).
 #include <functional>
 #include <vector>
 
 namespace traccc {
 
-using position_estimation = std::function<vector2(channel_id, channel_id)>;
-using signal_modeling = std::function<scalar(scalar)>;
+/// Function for signal modelling
+inline scalar signal_cell_modelling(scalar signal_in) {
+    return signal_in;
+}
+
+/// Function for pixel segmentation
+inline vector2 position_from_cell(cell c, pixel_data pixel, bool is_default) {
+    if (is_default)
+        return {static_cast<scalar>(c.channel0),
+                static_cast<scalar>(c.channel1)};
+    else
+        // Retrieve the specific values based on module idx (for now hard-coded)
+        return {pixel.min_center_x + c.channel0 * pixel.pitch_x,
+                pixel.min_center_y + c.channel1 * pixel.pitch_y};
+}
 
+inline vector2 get_pitch(pixel_data pixel) {
+    // return the values based on the module idx (for now hard-coded)
+    return {pixel.pitch_x, pixel.pitch_y};
+}
+
+using position_estimation = std::function<vector2(channel_id, channel_id)>;
 struct cluster_id {
 
     event_id event = 0;
+    scalar module_idx = 0;
     geometry_id module = 0;
     transform3 placement = transform3{};
-
-    position_estimation position_from_cell = [](channel_id ch0,
-                                                channel_id ch1) -> vector2 {
-        return {static_cast<scalar>(ch0), static_cast<scalar>(ch1)};
-    };
-
     scalar threshold = 0.;
-    signal_modeling signal = [](scalar signal_in) -> scalar {
-        return signal_in;
-    };
+
+    pixel_data pixel;
+    bool is_default = true;
```

Comment:
Could you explain a bit what this `is_default` bit is for? :thinking: I see that you use it, but do we really need a separate variable for it?

Reply:
So as far as I understand, if the "whole" clusterization algorithm is called (with component connection -> measurement creation - > spacepoint formation), the `position_estimation()` function returns the `vector2` based on the pixel_data. So my understanding before was that the "default" behaviour of this function which was previously:
 ```cpp
position_estimation position_from_cell = [](channel_id ch0,
                                                channel_id ch1) -> vector2 {
        return {static_cast<scalar>(ch0), static_cast<scalar>(ch1)};
    };
```
is basically never called.
But I saw that this is not true and it is called in some cases - in some of the tests. I suppose that those tests are running just measurement creation on its own and that's when the default behaviour is called. 

So this bit is basically always switched to false if the whole clusterization algorithm is called but otherwise it's true - and this guides the behaviour of the `position_estimation` function. 

Not sure if that makes sense... :stuck_out_tongue: I'm aware that this is not ideal since from what I understand, in the end this would be removed and position estimation would be just: 
```cpp
inline vector2 position_from_cell(cell c, pixel_data pixel) {
    return {pixel.min_center_x + c.channel0 * pixel.pitch_x,
            pixel.min_center_y + c.channel1 * pixel.pitch_y};
```
instead of:
```cpp
inline vector2 position_from_cell(cell c, pixel_data pixel, bool is_default) {
 if (is_default)
        return {static_cast<scalar>(c.channel0),
                static_cast<scalar>(c.channel1)};
    else
        // Retrieve the specific values based on module idx (for now hard-coded)
        return {pixel.min_center_x + c.channel0 * pixel.pitch_x,
                pixel.min_center_y + c.channel1 * pixel.pitch_y};
}
```
But I haven't found any other way to work around it

Reply:
oh of course, that's true...  :stuck_out_tongue: I changed it now.